src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1253,3 +1253,104 @@ def get_parent_proc_name():
         parent_proc_name = _get_parent_proc_name()
         setattr(get_parent_proc_name, "return_value", parent_proc_name)
     return getattr(get_parent_proc_name, "return_value")
+
+
+def log_cmd_history(command, args):
+    import os
+    from knack.util import ensure_dir
+    from azure.cli.core.extension import get_extension, ExtensionNotInstalledException
+    from azure.cli.core._environment import get_config_dir
+
+    if command == 'next':
+        return
+
+    # Determine whether "az next" has been installed.
+    # At present, command execution log is only recorded when "az next" is installed
+    try:
+        az_next_is_installed = get_extension("next")
+    except ExtensionNotInstalledException:
+        az_next_is_installed = False
+
+    if not az_next_is_installed:
+        return
+
+    # For commands that do not require logging, we can pass in the parameter "--no-log" to control
+    if '--no-log' in args:
+        return
+
+    if not args or not command:
+        return
```

Comment:
Can we combine these early return conditions before determining whether "az next" has been installed as `get_extension` cost a lot.
```suggestion
    if not args or not command or command == 'next' or '--no-log' in args:
        return

    # Determine whether "az next" has been installed.
    # At present, command execution log is only recorded when "az next" is installed
    try:
        get_extension("next")
    except ExtensionNotInstalledException:
        return
```